### PR TITLE
Fix accidental closure in Regex.Replace

### PR DESCRIPTION
**Pull request by stephentoub**
And sprinkle some "static" around to help avoid this in the future.

### CODE REVIEW

File: src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs [hunk 1]
```diff
@@ -193,11 +193,11 @@ private static string Replace(MatchEvaluator evaluator, Regex regex, string inpu
             {
                 state.prevat = input.Length;
 
-                regex.Run(input, startat, ref state, (ref (SegmentStringBuilder segments, MatchEvaluator evaluator, int prevat, string input, int count) state, Match match) =>
+                regex.Run(input, startat, ref state, static (ref (SegmentStringBuilder segments, MatchEvaluator evaluator, int prevat, string input, int count) state, Match match) =>
                 {
                     state.segments.Add(state.input.AsMemory(match.Index + match.Length, state.prevat - match.Index - match.Length));
                     state.prevat = match.Index;
-                    state.segments.Add(evaluator(match).AsMemory());
+                    state.segments.Add(state.evaluator(match).AsMemory());
```
Comment (stephentoub) [inline src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs:200]: This is the fix.

Comment (pgovind) [reply src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs:200]: Just curious about how you found this?

Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs:200]: Wednesdays are a busy meeting day for me, and I find my mind is able to stay more focused on them if I give it little mindless side projects, like walking in ILSpy through all "DisplayClass"es in all of netcoreapp looking for ones that don't belong :)

**Review by pgovind — APPROVED** [review]
Pretty straightforward. LGTM!

**Review by eerhardt — APPROVED** [review]
LGTM

**Comment by iSazonov** [comment]
The use of "static" is not so obvious.  Do you already have an idea to make an analyzer for this?

File: src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs (hunk 1 above)
Comment (davidfowl) [reply src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs:200]: Next task is to look a dumps all day.

**Pull request merged**